string.go

```diff
@@ -30,40 +28,45 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts text and given key with AES.
-func AESEncrypt(key, text []byte) ([]byte, error) {
+// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
```

Comment:
Encryption:

1. Read the 96-bit nonce with `crypto/rand`
2. Use it to encrypt your stuff
3. Return `nonce || ciphertext` where `||` stands for concatenation

Decryption:
1. Extract the nonce from the "ciphertext" (first 96 bits)
2. Use it to decrypt the ciphertext
3. Return the plaintext

Even though the nonce is only 12 bytes long, creating it randomly instead of using a counter is no problem unless you intend to encrypt more than 2^32 messages using the same key.